platform/ui/src/contextProviders/LabellingFlowProvider.js

```diff
@@ -0,0 +1,131 @@
+  useEffect,
+  useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+
+import LabellingManager from '../../../viewer/src/components/Labelling/LabellingManager';
+
+import { useDialog } from './DialogProvider';
+
+const LabellingFlowContext = createContext(null);
+const { Provider } = LabellingFlowContext;
+
+export const useLabellingFlow = () => useContext(LabellingFlowContext);
+
+const LabellingFlowProvider = ({ children, service, commandsManager }) => {
+  const { create, dismiss } = useDialog();
+
+  /**
+   * Sets the implementation of a labelling flow service that can be used by extensions.
+   *
+   * @returns void
+   */
+  useEffect(() => {
+    if (service) {
+      service.setServiceImplementation({
+        show,
+        hide,
+      });
+    }
+  }, [hide, service, show]);
+
+  const hide = useCallback(() => dismiss({ id: 'labelling' }), [dismiss]);
+
+  const show = useCallback(
+    ({ centralize, defaultPosition, props }) => {
+      hide();
+      create({
+        id: 'labelling',
+        centralize,
+        isDraggable: false,
+        showOverlay: true,
+        content: LabellingManager,
+        defaultPosition,
+        contentProps: {
+          visible: true,
+          measurementData: props.measurementData,
+          labellingDoneCallback: () => dismiss({ id: 'labelling' }),
+          updateLabelling: labellingData =>
+            _updateLabellingHandler(labellingData, props.measurementData),
+          ...props,
+        },
+      });
+    },
+    [_updateLabellingHandler, create, dismiss, hide]
+  );
+
+  const _updateLabellingHandler = useCallback(
+    (labellingData, measurementData) => {
+      const { location, description, response } = labellingData;
+
+      if (location) {
+        measurementData.location = location;
+      }
+
+      measurementData.description = description || '';
+
+      if (response) {
+        measurementData.response = response;
+      }
+
+      commandsManager.runCommand(
+        'updateTableWithNewMeasurementData',
+        measurementData
+      );
+    },
+    [commandsManager]
+  );
+
+  return (
+    <Provider
+      value={{
+        show,
+        hide,
+      }}
+    >
+      {children}
+    </Provider>
+  );
+};
+
+/**
+ * Higher Order Component to use the labelling flow methods through a Class Component.
+ *
+ * @returns
+ */
+export const withLabellingFlow = Component => {
+  return function WrappedComponent(props) {
+    const { show, hide } = useLabellingFlow();
+    return (
+      <Component
+        {...props}
+        modal={{
+          show,
+          hide,
+        }}
+      />
+    );
+  };
+};
+
+LabellingFlowProvider.defaultProps = {
+  service: null,
+};
+
+LabellingFlowProvider.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+  service: PropTypes.shape({
+    setServiceImplementation: PropTypes.func,
+  }),
+  commandsManager: PropTypes.object.isRequired,
```

Comment:
Instead of taking in the entire commands manager, have this be a callback method? `onSelection` or something. Then, in the props for the provider, set the value as:

```js
onSelect={(val) => {
      commandsManager.runCommand(
        'updateTableWithNewMeasurementData',
        measurementData
      );
}}
```

Or something similar. Our generic UI shouldn't need to know or care about the commands manager

Reply:
Yeah but im afraid that the provider will have a bunch of callbacks being passed through App.js. Do you have any different alternative in mind?